manifests/plugin/container_gateway.pp

```diff
@@ -22,13 +36,33 @@
   Boolean $enabled = true,
   Foreman_proxy::ListenOn $listen_on = 'https',
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
+  Enum['postgres', 'sqlite'] $database_backend = 'postgres',
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Boolean $manage_postgresql = true,
+  Stdlib::Host $postgresql_host = 'localhost',
+  Stdlib::Port $postgresql_port = 5432,
```

Comment:
I'd be tempted to leave these `undef` by default and prefer a UNIX socket. I think if you use `postgresql:///container_gateway` as the URL that it will do that. https://www.cybertec-postgresql.com/en/postgresql-performance-advice-unix-sockets-vs-localhost/ is from 2016 and shows a 60% performance increase. Other sources show a more conservative 15% or 30%. Still, free performance increase isn't a bad idea.

Reply:
Interesting, I'll have to test this out.

Reply:
Confirmed it works leaving ouy the host and port.